core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -580,23 +580,26 @@ public PlanOptimizers(
                                 new PruneTableScanColumns(metadata),
```

Comment:
Move "constraints" from commit message to code where rule is executed? This way rule ordering constraints won't be "lost" when rule line is overriden. It's also much easier to read javadoc in `PlanOptimizers` then search for explanation in commits

Reply:
Copied from commit into comments.